src/modules/logicModule/BountyManager.sol

```diff
@@ -0,0 +1,528 @@
+        address sender = _msgSender();
+        Contributor[] memory contribs = _claimRegistry[claimId].contributors;
+        uint length = contribs.length;
+        bool found;
+        for (uint i; i < length; i++) {
+            if (contribs[i].addr == sender) {
+                found = true;
+                break;
+            }
+        }
+
+        if (!found) {
+            revert Module__BountyManager__OnlyClaimContributor();
+        }
+        _;
+    }
+
+    modifier validPayoutAmounts(
+        uint minimumPayoutAmount,
+        uint maximumPayoutAmount
+    ) {
+        if (
+            minimumPayoutAmount == 0
+                || maximumPayoutAmount < minimumPayoutAmount
+        ) {
+            revert Module__BountyManager__InvalidPayoutAmounts();
+        }
+        _;
+    }
+
+    modifier validBountyId(uint bountyId) {
+        if (!isExistingBountyId(bountyId)) {
+            revert Module__BountyManager__InvalidBountyId();
+        }
+        _;
+    }
+
+    modifier validClaimId(uint claimId) {
+        if (!isExistingClaimId(claimId)) {
+            revert Module__BountyManager__InvalidClaimId();
+        }
+        _;
+    }
+
+    function validContributorsForBounty(
+        Contributor[] memory contributors,
+        Bounty memory bounty
+    ) internal view {
+        //@update to be in correct range
+        uint length = contributors.length;
+        //length cant be zero
+        if (length == 0) {
+            revert Module__BountyManager__InvalidContributorsLength();
+        }
+        uint totalAmount;
+        uint currentAmount;
+        address contrib;
+        for (uint i; i < length; i++) {
+            currentAmount = contributors[i].claimAmount;
+
+            //amount cant be zero
+            if (currentAmount == 0) {
+                revert Module__BountyManager__InvalidContributorAmount();
+            }
+
+            contrib = contributors[i].addr;
+            if (
+                contrib == address(0) || contrib == address(this)
+                    || contrib == address(proposal())
+            ) {
+                revert Module__BountyManager__InvalidContributorAddress();
+            }
+
+            totalAmount += currentAmount;
+        }
+
+        if (
+            totalAmount > bounty.maximumPayoutAmount
+                || totalAmount < bounty.minimumPayoutAmount
+        ) {
+            revert Module__BountyManager__ClaimExceedsGivenPayoutAmounts();
+        }
+    }
+
+    modifier claimBelongingToBounty(uint claimId, uint bountyId) {
+        //Its not claimed if claimedBy is still 0
+        if (_claimRegistry[claimId].bountyId != bountyId) {
+            revert Module__BountyManager__ClaimNotBelongingToBounty();
+        }
+        _;
+    }
+
+    modifier notClaimed(uint bountyId) {
+        //Its not claimed if claimedBy is still 0
+        if (_bountyRegistry[bountyId].claimedBy != 0) {
+            revert Module__BountyManager__BountyAlreadyClaimed();
```

Comment:
This revert message will be thrown up even when the bounty is locked and the `claimedBy` value is type(unit).max, which will not be an entirely accurate error message.